docs/source/models/segment_anything.rst

```diff
@@ -0,0 +1,251 @@
+    Model (SAM) and corresponding dataset (SA-1B) of 1B masks and 11M images at https://segment-anything.com to foster
+    research into foundation models for computer vision.
+
+    **Tasks:** Segmentation
+
+    **Datasets:** SA-1B
+
+    **Licence:** Apache
+
+    +++
+    **Authors:** Alexander Kirillov and Eric Mintun and Nikhila Ravi and Hanzi Mao and Chloe Rolland and Laura
+    Gustafson and Alex Berg and Wan-Yen Lo and Piotr Dollar and Ross Girshick
+
+
+
+How to use SAM from Kornia
+--------------------------
+
+Load/build the model
+^^^^^^^^^^^^^^^^^^^^
+This is an example how you can load the model.
+
+.. code-block:: python
+
+    from kornia.contrib import Sam
+
+    # model_type can be:
+    #   0, 'vit_h' or `kornia.contrib.SamModelType.vit_h`
+    #   1, 'vit_l' or `kornia.contrib.SamModelType.vit_l`
+    #   2, 'vit_b' or `kornia.contrib.SamModelType.vit_b`
+    model_type = 'vit_b' # or can be a number `2` or the enum kornia.contrib.SamModelType.vit_b
+
+    # Load/build the model
+    sam_model = Sam.build(model_type)
+
+Load checkpoint
+^^^^^^^^^^^^^^^
+With the load checkpoint method you can load from a file or directly from a URL. The official (by meta) model weights are:
+
+#. `vit_h`: `ViT-H SAM model - https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth <https://dl.fbaipublicfiles.com/segment_anything/sam_vit_h_4b8939.pth>`_.
+#. `vit_l`: `ViT-L SAM model - https://dl.fbaipublicfiles.com/segment_anything/sam_vit_l_0b3195.pth <https://dl.fbaipublicfiles.com/segment_anything/sam_vit_l_0b3195.pth>`_.
+#. `vit_b`: `ViT-B SAM model - https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth <https://dl.fbaipublicfiles.com/segment_anything/sam_vit_b_01ec64.pth>`_.
+
+
+.. code-block:: python
+
+    from kornia.contrib import Sam
+    from kornia.utils import get_cuda_device_if_available
+
+    model_type = 'vit_b'
+
+    checkpoint = './path_for_the_model_checkpoint.pth' # Can be a filepath or a url
+    device = get_cuda_device_if_available()
+
+    # Load/build the model
+    sam_model = Sam.build(model_type)
+
+    # Load the checkpoint
+    sam_model.load_checkpoint(checkpoint, device)
+
+Load from pretrained
+^^^^^^^^^^^^^^^^^^^^
+This method internally uses `build` and `load_checkpoint`, also move the model for the desired device.
+
+.. code-block:: python
+
+    from kornia.contrib import Sam
+    from kornia.utils import get_cuda_device_if_available
+
+    model_type = 'vit_b'
+
+    checkpoint = './path_for_the_model_checkpoint.pth' # Can be a filepath or a url
+    device = get_cuda_device_if_available()
+
+    # Load the model with checkpoint on the desired device
+    sam_model = Sam.from_pretrained(model_type, checkpoint, device)
+
+
+
+Predictor
+^^^^^^^^^
+.. _anchor Predictor:
+
+The High level API `SamPrediction` handle with the image and prompt transformation, preprocessing and prediction for
```

Comment:
```suggestion
The High level API `SamPredictor` handle with the image and prompt transformation, preprocessing and prediction for
```
However I’m favour of starting already to rename it as ImagePrompter

Reply:
i added a first version of a prompter on https://github.com/kornia/kornia/pull/2315/commits/48eb9f9fdef37d423cee292652fc1a84925f1a9d

not sure if I liked it or not xD